modules/hitl2/src/backend/api.ts

```diff
@@ -0,0 +1,289 @@
+    }
+  }
+
+  router.get(
+    '/agents/me',
+    hitlMiddleware(async (req: RequestWithUser, res: Response) => {
+      const { email, strategy } = req.tokenUser!
+      const payload = await repository.getCurrentAgent(req, req.params.botId, makeAgentId(strategy, email))
+      res.json(payload)
+    })
+  )
+
+  router.get(
+    '/agents',
+    hitlMiddleware(async (req: RequestWithUser, res: Response) => {
+      const agents = await repository.getAgents(
+        req.params.botId,
+        _.tap(_.pick(req.query, 'online'), conditions => {
+          if (conditions.online) {
+            conditions.online = boolean(conditions.online)
+          }
+        }) as AgentCollectionConditions
+      )
+      res.json(agents)
+    })
+  )
+
+  router.post(
+    '/agents/me/online',
+    hitlMiddleware(async (req: RequestWithUser, res: Response) => {
+      const { email, strategy } = req.tokenUser!
+      const agentId = makeAgentId(strategy, email)
+
+      const online = await repository.setAgentOnline(req.params.botId, agentId, true)
+      const payload = {
+        online: online
+      }
+
+      realtime.sendPayload({
+        resource: 'agent',
+        type: 'update',
+        id: agentId,
+        payload: payload
+      })
+
+      res.json(payload)
+    })
+  )
+
+  router.post(
+    '/agents/me/offline',
+    hitlMiddleware(async (req: RequestWithUser, res: Response) => {
+      const { email, strategy } = req.tokenUser!
+      const agentId = makeAgentId(strategy, email)
+
+      const online = await repository.setAgentOnline(req.params.botId, agentId, false)
+      const payload = {
+        online: online
+      }
+
+      realtime.sendPayload({
+        resource: 'agent',
+        type: 'update',
+        id: agentId,
+        payload: payload
+      })
+
+      res.json(payload)
+    })
+  )
+
+  router.get(
+    '/escalations',
+    hitlMiddleware(async (req: Request, res: Response) => {
+      const escalations = await repository.getEscalationsWithComments(
+        req.params.botId,
+        _.pick(req.query, ['limit', 'orderByColumn', 'orderByDirection']) as CollectionConditions
+      )
+      res.json(escalations)
+    })
+  )
+
+  router.post(
+    '/escalations',
+    hitlMiddleware(async (req: Request, res: Response) => {
+      const payload = {
+        ..._.pick(req.body, ['userId', 'userThreadId']),
+        status: 'pending' as 'pending'
+      }
+
+      Joi.attempt(payload, CreateEscalationSchema)
+
+      // Prevent creating a new escalation if one is currently pending or assigned
+      let escalation
+      escalation = await repository
+        .escalationsQuery(builder => {
+          return builder
+            .where('botId', req.params.botId)
+            .andWhere('userId', payload.userId)
+            .andWhere('userThreadId', payload.userThreadId)
+            .whereNot('status', 'resolved')
+            .orderBy('createdAt')
+            .limit(1)
+        })
+        .then(data => _.head(data))
```

Comment:
Can you keep all the SQL logic inside `repository.ts` ? Also, please name that file `db.ts` so it's consistent with other modules